pyrtl/inputoutput.py

```diff
@@ -669,9 +669,12 @@ def _to_verilog_combinational(self):
                 print('    assign %s = {%s};' % t, file=self.file)
             elif net.op == 'r':
                 pass  # do nothing for registers
-            elif net.op == 'm':
+            elif net.op == 'm':  # use always block and assign as Verilog register
+                print('    always @( posedge clk )', file=self.file)
+                print('    begin', file=self.file)
                 t = (self._varname(net.dests[0]), net.op_param[0], self._varname(net.args[0]))
-                print('        assign %s = mem_%s[%s];' % t, file=self.file)
+                print('        %s <= mem_%s[%s];' % t, file=self.file)
```

Comment:
@masoug I changed part of the header where the wire should have been a reg, but it is hard for me to tell whether that will solve the whole problem or just one instance of it.